### ISSUE TITLE
Cannot create users in 2.5.10

### ISSUE THREAD

**Issue by infocynic** [n2cms/n2cms#463]
Trying to create any users results in a NullReferenceException as per stack trace below:

Stack Trace: 

```
[NullReferenceException: Object reference not set to an instance of an object.]
   N2.Security.<>c__DisplayClass8.<GetUserNameByEmail>b__5(ContentItem x) in c:\Projekt\OSS\github\n2cms\n2cms\src\Mvc\MvcTemplates\N2\Security\ContentMembershipProvider.cs:423
   System.Linq.WhereSelectListIterator`2.MoveNext() +172
   System.Linq.Enumerable.FirstOrDefault(IEnumerable`1 source) +164
   N2.Security.ContentMembershipProvider.GetUserNameByEmail(String email) in c:\Projekt\OSS\github\n2cms\n2cms\src\Mvc\MvcTemplates\N2\Security\ContentMembershipProvider.cs:425
   N2.Security.ContentMembershipProvider.CreateUser(String username, String password, String email, String passwordQuestion, String passwordAnswer, Boolean isApproved, Object providerUserKey, MembershipCreateStatus& status) in c:\Projekt\OSS\github\n2cms\n2cms\src\Mvc\MvcTemplates\N2\Security\ContentMembershipProvider.cs:269
   System.Web.UI.WebControls.CreateUserWizard.AttemptCreateUser() +304
```

(rest of trace removed, all system code)

When I switch to my standby membership provider, which inherits the standard N2 one and overrides methods, specifically `GetUserNameByEmail`, it works correctly.

My `GetUserNameByEmail` uses this logic:

```
var userNames = users.Where(x => x.Details["Email"].ToString() == email).Select(x => x.Name);
```

The current version on trunk (I assume this is pretty close to 2.5.10; I haven't done a full checkout/compare) is:

```
var userNames = users.Where(x => x.Details.ContainsKey("Email") && 
x.Details["Email"].ToString().Equals(email, StringComparison.OrdinalIgnoreCase)).Select(x => x.Name);
```

I have a case where `x.Details["Email"]` is null, but `x.Details.ContainsKey("Email")` is true.. You cannot then invoke `String.Equals` on the null object. 

Now, it's certainly a fair question "how did you get a user with a null e-mail address?" Pretty easily, actually, as nothing in `CreateUser` requires it. (My override does, to prevent this case, but for whatever reason, the default N2 handling does not.)

TL;DR:
If you have a null e-mail address for any user, which is possible under default N2 settings, you will get an exception under default N2 settings when you create a new user. The fix requires slightly rewriting `GetUserNameByEmail` to trap for this case.

```
var userNames = users.Where(x => x.Details.ContainsKey("Email") && 
!String.IsNullOrEmpty(x.Details["Email"].ToString()) && 
x.Details["Email"].ToString().Equals(email, StringComparison.OrdinalIgnoreCase)).Select(x => x.Name);
```

I do not have time right not to create a pull request or test this fix; I have to go live with what I have already working, but I can probably test a fix next week.
